### ISSUE TITLE
Build process fails with error about function ‘gss_store_cred_into’

### ISSUE THREAD

**Issue by sogalboogalsky** [pythongssapi/python-gssapi#131]
### What went wrong?
When trying to build from git or installing with "pip install", the process fails with the following output:

```
    gssapi/raw/ext_s4u.c: In function ‘__pyx_pf_6gssapi_3raw_7ext_s4u_acquire_cred_impersonate_name’:
    gssapi/raw/ext_s4u.c:1168:9: warning: implicit declaration of function ‘gss_acquire_cred_impersonate_name’ [-Wimplicit-function-declaration]
             __pyx_v_maj_stat = gss_acquire_cred_impersonate_name((&__pyx_v_min_stat), __pyx_v_impersonator_cred->raw_creds, __pyx_v_name->raw_name, __pyx_v_input_ttl, __pyx_v_desired_mechs, __pyx_v_c_usage, (&__pyx_v_creds), (&__pyx_v_actual_mechs), (&__pyx_v_actual_ttl));
             ^
    gssapi/raw/ext_s4u.c: In function ‘__pyx_pf_6gssapi_3raw_7ext_s4u_2add_cred_impersonate_name’:
    gssapi/raw/ext_s4u.c:1735:9: warning: implicit declaration of function ‘gss_add_cred_impersonate_name’ [-Wimplicit-function-declaration]
             __pyx_v_maj_stat = gss_add_cred_impersonate_name((&__pyx_v_min_stat), __pyx_v_raw_input_cred, __pyx_v_impersonator_cred->raw_creds, __pyx_v_name->raw_name, (&__pyx_v_mech->raw_oid), __pyx_v_c_usage, __pyx_v_input_initiator_ttl, __pyx_v_input_acceptor_ttl, (&__pyx_v_creds), (&__pyx_v_actual_mechs), (&__pyx_v_actual_initiator_ttl), (&__pyx_v_actual_acceptor_ttl));
             ^
    gcc -pthread -shared build/temp.linux-x86_64-3.4/gssapi/raw/ext_s4u.o -L/usr/lib64 -lpython3.4m -o build/lib.linux-x86_64-3.4/gssapi/raw/ext_s4u.cpython-34m.so -lgssapi_krb5 -lkrb5 -lk5crypto -lcom_err
    building 'gssapi.raw.ext_cred_store' extension
    gcc -pthread -Wno-unused-result -DNDEBUG -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector -funwind-tables -fasynchronous-unwind-tables -g -DOPENSSL_LOAD_CONF -fPIC -I/usr/include/python3.4m -c gssapi/raw/ext_cred_store.c -o build/temp.linux-x86_64-3.4/gssapi/raw/ext_cred_store.o
    gssapi/raw/ext_cred_store.c:788:1: error: unknown type name ‘gss_key_value_set_desc’
     static gss_key_value_set_desc *__pyx_f_6gssapi_3raw_14ext_cred_store_c_create_key_value_set(PyObject *); /*proto*/
     ^
    gssapi/raw/ext_cred_store.c:789:72: error: unknown type name ‘gss_key_value_set_desc’
     static void __pyx_f_6gssapi_3raw_14ext_cred_store_c_free_key_value_set(gss_key_value_set_desc *); /*proto*/
                                                                            ^
    gssapi/raw/ext_cred_store.c:942:1: error: unknown type name ‘gss_key_value_set_desc’
     static gss_key_value_set_desc *__pyx_f_6gssapi_3raw_14ext_cred_store_c_create_key_value_set(PyObject *__pyx_v_values) {
     ^
    gssapi/raw/ext_cred_store.c: In function ‘__pyx_f_6gssapi_3raw_14ext_cred_store_c_create_key_value_set’:
    gssapi/raw/ext_cred_store.c:943:3: error: unknown type name ‘gss_key_value_set_desc’
       gss_key_value_set_desc *__pyx_v_res;
       ^
    gssapi/raw/ext_cred_store.c:947:3: error: unknown type name ‘gss_key_value_set_desc’
       gss_key_value_set_desc *__pyx_r;
       ^
    gssapi/raw/ext_cred_store.c:974:19: error: ‘gss_key_value_set_desc’ undeclared (first use in this function)
       __pyx_v_res = ((gss_key_value_set_desc *)malloc((sizeof(gss_key_value_set_desc))));
                       ^
    gssapi/raw/ext_cred_store.c:974:19: note: each undeclared identifier is reported only once for each function it appears in
    gssapi/raw/ext_cred_store.c:974:43: error: expected expression before ‘)’ token
       __pyx_v_res = ((gss_key_value_set_desc *)malloc((sizeof(gss_key_value_set_desc))));
                                               ^
    gssapi/raw/ext_cred_store.c:1020:14: error: request for member ‘count’ in something not a structure or union
       __pyx_v_res->count = __pyx_t_3;
                  ^
    gssapi/raw/ext_cred_store.c:1029:14: error: request for member ‘elements’ in something not a structure or union
       __pyx_v_res->elements = ((gss_key_value_element_desc *)calloc(__pyx_v_res->count, (sizeof(gss_key_value_element_desc))));
                  ^
    gssapi/raw/ext_cred_store.c:1029:29: error: ‘gss_key_value_element_desc’ undeclared (first use in this function)
       __pyx_v_res->elements = ((gss_key_value_element_desc *)calloc(__pyx_v_res->count, (sizeof(gss_key_value_element_desc))));
                                 ^
    gssapi/raw/ext_cred_store.c:1029:57: error: expected expression before ‘)’ token
       __pyx_v_res->elements = ((gss_key_value_element_desc *)calloc(__pyx_v_res->count, (sizeof(gss_key_value_element_desc))));
                                                             ^
    gssapi/raw/ext_cred_store.c:1038:28: error: request for member ‘elements’ in something not a structure or union
       __pyx_t_1 = ((__pyx_v_res->elements == NULL) != 0);
                                ^
    gssapi/raw/ext_cred_store.c:1189:17: error: request for member ‘elements’ in something not a structure or union
         (__pyx_v_res->elements[__pyx_t_12]).key = __pyx_t_11;
                     ^
    gssapi/raw/ext_cred_store.c:1200:17: error: request for member ‘elements’ in something not a structure or union
         (__pyx_v_res->elements[__pyx_t_12]).value = __pyx_t_13;
                     ^
    gssapi/raw/ext_cred_store.c: At top level:
    gssapi/raw/ext_cred_store.c:1257:72: error: unknown type name ‘gss_key_value_set_desc’
     static void __pyx_f_6gssapi_3raw_14ext_cred_store_c_free_key_value_set(gss_key_value_set_desc *__pyx_v_kvset) {
                                                                            ^
    gssapi/raw/ext_cred_store.c: In function ‘__pyx_pf_6gssapi_3raw_14ext_cred_store_acquire_cred_from’:
    gssapi/raw/ext_cred_store.c:1425:3: error: unknown type name ‘gss_key_value_set_desc’
       gss_key_value_set_desc *__pyx_v_c_store;
       ^
    gssapi/raw/ext_cred_store.c:1439:3: error: unknown type name ‘gss_key_value_set_desc’
       gss_key_value_set_desc *__pyx_t_5;
       ^
    gssapi/raw/ext_cred_store.c:1656:23: error: ‘GSS_C_NO_CRED_STORE’ undeclared (first use in this function)
         __pyx_v_c_store = GSS_C_NO_CRED_STORE;
                           ^
    gssapi/raw/ext_cred_store.c:1681:9: warning: implicit declaration of function ‘gss_acquire_cred_from’ [-Wimplicit-function-declaration]
             __pyx_v_maj_stat = gss_acquire_cred_from((&__pyx_v_min_stat), __pyx_v_c_name, __pyx_v_input_ttl, __pyx_v_desired_mechs, __pyx_v_c_usage, __pyx_v_c_store, (&__pyx_v_creds), (&__pyx_v_actual_mechs), (&__pyx_v_actual_ttl));
             ^
    gssapi/raw/ext_cred_store.c:1749:5: warning: implicit declaration of function ‘__pyx_f_6gssapi_3raw_14ext_cred_store_c_free_key_value_set’ [-Wimplicit-function-declaration]
         __pyx_f_6gssapi_3raw_14ext_cred_store_c_free_key_value_set(__pyx_v_c_store);
         ^
    gssapi/raw/ext_cred_store.c: In function ‘__pyx_pf_6gssapi_3raw_14ext_cred_store_2add_cred_from’:
    gssapi/raw/ext_cred_store.c:2087:3: error: unknown type name ‘gss_key_value_set_desc’
       gss_key_value_set_desc *__pyx_v_c_store;
       ^
    gssapi/raw/ext_cred_store.c:2102:3: error: unknown type name ‘gss_key_value_set_desc’
       gss_key_value_set_desc *__pyx_t_6;
       ^
    gssapi/raw/ext_cred_store.c:2307:23: error: ‘GSS_C_NO_CRED_STORE’ undeclared (first use in this function)
         __pyx_v_c_store = GSS_C_NO_CRED_STORE;
                           ^
    gssapi/raw/ext_cred_store.c:2332:9: warning: implicit declaration of function ‘gss_add_cred_from’ [-Wimplicit-function-declaration]
             __pyx_v_maj_stat = gss_add_cred_from((&__pyx_v_min_stat), __pyx_v_c_input_creds, __pyx_v_c_name, __pyx_v_c_mech, __pyx_v_c_usage, __pyx_v_input_initiator_ttl, __pyx_v_input_acceptor_ttl, __pyx_v_c_store, (&__pyx_v_creds), (&__pyx_v_actual_mechs), (&__pyx_v_actual_initiator_ttl), (&__pyx_v_actual_acceptor_ttl));
             ^
    gssapi/raw/ext_cred_store.c: In function ‘__pyx_pf_6gssapi_3raw_14ext_cred_store_4store_cred_into’:
    gssapi/raw/ext_cred_store.c:2716:3: error: unknown type name ‘gss_key_value_set_desc’
       gss_key_value_set_desc *__pyx_v_c_store;
       ^
    gssapi/raw/ext_cred_store.c:2727:3: error: unknown type name ‘gss_key_value_set_desc’
       gss_key_value_set_desc *__pyx_t_3;
       ^
    gssapi/raw/ext_cred_store.c:2892:23: error: ‘GSS_C_NO_CRED_STORE’ undeclared (first use in this function)
         __pyx_v_c_store = GSS_C_NO_CRED_STORE;
                           ^
    gssapi/raw/ext_cred_store.c:2927:9: warning: implicit declaration of function ‘gss_store_cred_into’ [-Wimplicit-function-declaration]
             __pyx_v_maj_stat = gss_store_cred_into((&__pyx_v_min_stat), __pyx_v_c_creds, __pyx_v_c_usage, __pyx_v_desired_mech, __pyx_v_overwrite, __pyx_v_set_default, __pyx_v_c_store, (&__pyx_v_actual_mech_types), (&__pyx_v_actual_usage));
             ^
    error: command 'gcc' failed with exit status 1

```

### How do we reproduce?

On an up-to-date openSUSE Leap 42.3 system, launch command:
`pip install gssapi`
or
```
git clone https://github.com/pythongssapi/python-gssapi.git
python setup.py build
```
Both methods produces the same error

### Component versions (python-gssapi, Kerberos, OS / distro, etc.)

Kerberos components installed:

- krb5-devel 1.15.2 MIT Kerberos5
- krb5 1.15.2 MIT Kerberos5

OS is openSUSE Leap 42.3 64 bits, up-to-date

Python is 3.4.6 and 2.7.13 (same version for their respective _devel_ packages)

python-Cython is 0.23.4
python3-Cython is 0.24-3

cyrus-sasl-gssapi is 2.1.26

Thanks a lot.

**Comment by simo5** [comment]
Sounds like a krb5-config issue in Suse's Leap

From IRC:
```
krb5-config --libs gssapi
-lgssapi_krb5 -lkrb5 -lk5crypto -lcom_err
krb5-config --cflags gssapi
krb5-config gssapi --prefix
/usr/lib/mit
```
And there is no include/gssapi in /usr/lib/mit

**Comment by simo5** [comment]
I am closing for now, if it turns out we can handle the test better in setup.py then we'll open another issue.
thanks for reporting.
